l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -463,11 +504,16 @@ contract MailboxFacet is ZKChainBase, IMailboxImpl, MessageVerification {
 
         _writePriorityOp(transaction, _params.request.factoryDeps, canonicalTxHash, _params.expirationTimestamp);
         if (s.settlementLayer != address(0)) {
+            address assetRouter = IBridgehub(s.bridgehub).assetRouter();
+            bool getBalanceChange = _params.request.sender == AddressAliasHelper.applyL1ToL2Alias(assetRouter);
+
             // slither-disable-next-line unused-return
-            IMailbox(s.settlementLayer).requestL2TransactionToGatewayMailbox({
+            IMailbox(s.settlementLayer).requestL2TransactionToGatewayMailboxWithBalanceChange({
                 _chainId: s.chainId,
                 _canonicalTxHash: canonicalTxHash,
-                _expirationTimestamp: _params.expirationTimestamp
+                _expirationTimestamp: _params.expirationTimestamp,
+                _baseTokenAmount: _params.request.mintValue,
```

Comment:
It is okay that we trust the calling chain for the mintValue?

In theory a malicious calling chain could provide invalid value there.

Reply:
It is okay as long as dont have untrusted CTMs settling on top of chains, I'll just leave a comment about it in my PR